service_implementation/advanced_plm.md

```diff
@@ -1 +1,455 @@
-# Beyond just files: Using PLM for versioning and metadata
\ No newline at end of file
+# Beyond just files: Using PLM for versioning and metadata
+This document describes how to use PLM services from _inside_ a service, i.e., by calling the PLM API. The text includes  example code.
+There are two types of PLM services for use in CAxMan; both are described below:
+1) those that are integrated in the GSS framework;
+2) the PLM Server native web-services.
+
+## GSS
+
+The PLM Server has its own implementation of the GSS services. GSS offers SOAP and REST services:
+- WSDL (SOAP): https://caxman.clesgo.net/jotne/GSS/FileWebService?wsdl
+- REST end point: https://caxman.clesgo.net/jotne/GSS/resources/plm

```

Comment:
I'm not very happy with having URLs explicitly exposed like this. GSS offers the SOAP method `getDirectInteractionEndpoint(gss_URI)` which returns specifically the URLs above. These URLs should never be used directly (as in hard-coded) in any service and thus I don't think they should appear here.

It would also be very helpful here to see a short explanation here of how to obtain these endpoint URLs with the tools given to every service (namely, the gss URL which is in the `extraParameters` argument).

Reply:
Is there any description of the method 'getDirectInteractionEndpoint so reference could be given.
And I have very little knowledge about 'extraParameters'. Where I can read about that?

Reply:
No, there is not API reference of GSS yet, unfortunately. But the function expects a gss URI and returns a list of `EndpointInformation` objects, each containing the attributes `url` (string) and `type`(`["REST","SOAP"]`).

You can find some information on the extra parameters in this tutorial and especially in the code example:
https://github.com/CAxMan/docs-and-training/blob/master/tutorials/services/python_imageconverter.md